File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/AgentStatus.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum AgentStatus {
    Pending,
    Running,
    Failed;

    public String toWire() {
        return switch (this) {
            case Pending -> "pending";
            case Running -> "running";
            case Failed -> "failed";
        };
    }

    public static AgentStatus fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "pending" -> Pending;
            case "running" -> Running;
            case "failed" -> Failed;
            default -> throw new IllegalArgumentException("Unknown agent status: " + value);
        };
    }
}

File: apps/backend/src/Common/Domain/ValueObjects/AgentStatus.cs
namespace OffceOs.Domain.Common.ValueObjects;

public enum AgentStatus
{
    Pending,
    Running,
    Failed,
}

public static class AgentStatusExtensions
{
    public static string ToStorageString(this AgentStatus status) => status switch
    {
        AgentStatus.Pending => "pending",
        AgentStatus.Running => "running",
        AgentStatus.Failed => "failed",
        _ => throw new ArgumentOutOfRangeException(nameof(status)),
    };

    public static AgentStatus ToAgentStatus(this string value) => value switch
    {
        "pending" => AgentStatus.Pending,
        "running" => AgentStatus.Running,
        "failed" => AgentStatus.Failed,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown agent status: {value}"),
    };
}
